sources/packages/api/src/services/education-program-offering/education-program-offering.service.ts

```diff
@@ -95,4 +98,94 @@ export class EducationProgramOfferingService extends RecordDataModelService<Educ
       return item;
     });
   }
+
+  /**
+   * This is to fetch the Education Offering
+   * that are associated with the Location, Program
+   * and offering
+   * @param locationId
+   * @param programId
+   * @param offeringId
+   * @returns
+   */
+  async getProgramOffering(
+    locationId: number,
+    programId: number,
+    offeringId: number,
+  ): Promise<ProgramOfferingModel> {
+    return this.repo
+      .createQueryBuilder("offerings")
+      .select([
+        "offerings.id",
+        "offerings.name",
+        "offerings.studyStartDate",
+        "offerings.studyEndDate",
+        "offerings.breakStartDate",
+        "offerings.breakEndDate",
+        "offerings.actualTuitionCosts",
+        "offerings.programRelatedCosts",
+        "offerings.mandatoryFees",
+        "offerings.exceptionalExpenses",
+        "offerings.tuitionRemittanceRequestedAmount",
+        "offerings.offeringDelivered",
+        "offerings.lacksStudyDates",
+        "offerings.lacksStudyBreaks",
+        "offerings.lacksFixedCosts",
+        "offerings.tuitionRemittanceRequested",
+      ])
+      .innerJoin("offerings.educationProgram", "educationProgram")
+      .innerJoin("offerings.institutionLocation", "institutionLocation")
+      .where(
+        "offerings.id= :offeringId and educationProgram.id = :programId and institutionLocation.id = :locationId",
+        {
+          offeringId: offeringId,
+          programId: programId,
+          locationId: locationId,
+        },
+      )
+      .getOne();
+  }
+
+  /**
+   * Creates a new education program offering at program level
+   * @param locationId location id to associate the new program offering.
+   * @param programId program id to associate the new program offering.
+   * @param educationProgramOffering Information used to create the program offering.
+   * @returns Education program offering created.
+   */
+  async updateEducationProgramOffering(
```

Comment:
Is the logic for insert/update different in a way that justifies the creation of 2 different methods? Maybe we can have just one method to d both.

Reply:
We may not know, it would be always better to create 2 different methods rather having one and adding and editing dto's/methods/services when there is a difference between create and update.

Reply:
As per our conversation, this will not be considered a blocker, but IMHO we are preparing the application for something that is not an actual requirement and does not seems to be something that will be coming, and, in case it comes, it will not be hard to change it. Either way, all good to keep it.